File: UVa/Contents/c032/c032.py
import sys

def isPerfect(num):
    total = 1
    for i in range(2, num):
        if num % i == 0:
            total += i
    return total

count = 0
flag = 0
for temp in sys.stdin:
    temp = [int(e) for e in temp.split()]

    if count == 0:
        print("PERFECTION OUTPUT")

    for n in temp:
        if n == 0:
            print("END OF OUTPUT", end="")
            flag = 1
            break

        test = isPerfect(n)

        if n == test and n != 1:
            print("%5d  PERFECT" % n)
        elif n < test:
            print("%5d  ABUNDANT" % n)
        else:
            print("%5d  DEFICIENT" % n)
    
    if flag == 1:
        break 

    count += 1      

File: UVa/Contents/c032/c032.cpp
#include<iostream>
#include<iomanip>

using namespace std;

int isPerfect(int);

int main(void)
{
	int n, temp;
	int count = 0;
	
	while(1)
	{
		cin >> n;
		
		if(!n)
		{
			cout << "END OF OUTPUT" << endl;
			break;
		}
		
		if(!count)
			cout << "PERFECTION OUTPUT" << endl;
		
		temp = isPerfect(n);
		
		if(n == temp && n != 1)
			cout << setw(5) << n << "  PERFECT" << endl;
		else if(n < temp)
			cout << setw(5) << n << "  ABUNDANT" << endl;
		else
			cout << setw(5) << n << "  DEFICIENT" << endl;
		
		count++;
	}
	
	return 0;
}

int isPerfect(int num)
{
	int sum = 1;
	
	for(int i = 2; i < num; i++)
			if(!(num % i))
				sum += i;
	return sum;
}
